### DISCUSSION TITLE
How to recursively remove directory in an archive?

### DISCUSSION THREAD

**Discussion by gbook** [rikyoz/bit7z#200] [General]
How can I recursively remove a directory and all of it's contents in an existing archive?

I imagine using the BitArchiveEditor class, but I'm not sure how to use it to recursively remove a directory and it's contents.

**Comment by rikyoz** [comment]
> How can I recursively remove a directory and all of it's contents in an existing archive?

Unfortunately there isn't a direct way to do this, and I think there should be, so I'll definitely be working on adding that option as soon as possible.

> I imagine using the BitArchiveEditor class, but I'm not sure how to use it to recursively remove a directory and it's contents.

For now, the only way to remove a folder and its contents is something similar to this:

```cpp
    std::vector<uint32_t> indexes;
    {
        BitArchiveReader reader{lib, "./multiple_items.zip", BitFormat::Zip};
        for (const auto& item : reader) {
            if (item.path().starts_with("folder")) {
                indexes.push_back(item.index());
            }
        }
    }
    BitArchiveEditor editor{lib, "./multiple_items.zip", BitFormat::Zip};
    for (uint32_t index : indexes) {
        editor.deleteItem(index);
    }
    editor.applyChanges();
```
